src/Form/index.tsx

```diff
@@ -688,19 +693,38 @@ function Form({
             clientInstance.submitCustom(castValues, false);
           }
 
-          // User is authenticating. auth hook will set the initial stepKey once auth has finished
-          if (authState.redirectAfterLogin) return;
+          // console.log('redirectAfterLogin return early?', {
+          //   redirectAfterLogin: authState.redirectAfterLogin,
+          //   session,
+          //   formSettings,
+          //   redirectUrl
+          // });
+
+          const onboardingExists = areThereOnboardingQuestions(
+            session.integrations
+          );
+          const isFormComplete = session.form_completed;
+
+          // can I drop onboardingExists?
+          if (isFormComplete && !onboardingExists && authState.authId)
```

Comment:
should probably make this a shared fn between here + sms verify. just would need to have optional pass in of redirectUrl, since this case can't rely on the CBs